config/jobs/image-pushing/k8s-staging-csi-gen.sh

```diff
@@ -0,0 +1,66 @@
+#! /bin/bash -e
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+output="$(dirname $0)/k8s-staging-csi.yaml"
+repos="
+csi-driver-host-path
+csi-test
+external-attacher
+external-provisioner
+external-resizer
+external-snapshotter
+livenessprobe
+node-driver-registrar
+"
+
+cat >"$output" <<EOF
+# Automatically generated by k8s-staging-csi-gen.sh.
+
+postsubmits:
+EOF
+
+for repo in $repos; do
+    cat >>"$output" <<EOF
+  kubernetes-csi/$repo:
+    - name: post-$repo-push-images
+      cluster: k8s-infra-prow-build-trusted
+      annotations:
+        testgrid-dashboards: sig-storage-csi-$repo
+      decorate: true
+      branches:
+        # For publishing canary images. Publishing canary images for release branches can
+        # be added later, but then will depend on which release branches in each repo have
+        # the necessary cloud build files.
+        - ^master$
+        # This is a regex for semver, from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string.
+        # This is okay for upcoming releases, but old releases will not have the necessary cloud build
+        # files and thus the job will fail.
+        - ^v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$
```

Comment:
Do we need the branches to be specific, or can we enable it on everything?

Reply:
We can enable it for all tags. Doing it for all branches makes less sense, because an arbitrary branch name (say, "my-experimental-feature-branch") will then be used as-is in the image tag, which means the image will be built once and then all future branch updates will trigger this error during image pushing:
https://github.com/kubernetes-csi/node-driver-registrar/blob/da1e2a455271eb7c693dbd9835f6419c82e2ee02/release-tools/build.make#L154-L160

Reply:
So consider this limitation for branches, we somehow need to distinguish here between "real" branches and tags. I kept the semver check because it seemed reasonable enforce that tags on images have that format.